docs/onclusterlayering-quickstart.md

```diff
@@ -0,0 +1,492 @@
+
+```bash
+#!/usr/bin/env bash
+
+# (required): Clone the global image pull secret into a secret called
+# "global-pull-secret-copy" within the MCO namespace. This will be used to pull
+# the base OS image:
+oc create secret docker-registry global-pull-secret-copy \
+  --namespace "openshift-machine-config-operator" \
+  --from-file=.dockerconfigjson=<(oc get secret/pull-secret -n openshift-config -o go-template='{{index .data ".dockerconfigjson" | base64decode}}')
+
+# (required for OCP): Clone the Red Hat entitlements certificate into the MCO
+# namespace. This will be used to access RHEL content that you are entitled to
+# access.
+oc create secret generic etc-pki-entitlement \
+  --namespace "openshift-machine-config-operator" \
+  --from-file=entitlement.pem=<(oc get secret/etc-pki-entitlement -n openshift-config-managed -o go-template='{{index .data "entitlement.pem" | base64decode }}') \
+  --from-file=entitlement-key.pem=<(oc get secret/etc-pki-entitlement -n openshift-config-managed -o go-template='{{index .data "entitlement-key.pem" | base64decode }}')
+
+# (optional): Create an ImageStream which will act as our image registry. This
+# step can be omitted if you are using an external image registry such as
+# Quay.io, though you will need to create the push and pull secrets in Quay.io
+# as well as inside your cluster.
+oc create imagestream os-images -n openshift-machine-config-operator
+
+# (optional): To get the image registry pullspec for your ImageStream, you can run the following command:
+# This will return "image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/os-images"
+oc get imagestream/os-images  -n openshift-machine-config-operator -o=jsonpath='{.status.dockerImageRepository}'
+
+# (optional): Since this walk-through will use an ImageStream, we need to get the push and
+# pull secret associated with the builder service account. For the sake of this
+# demonstration, lets assume that this command returns the name
+# "builder-dockercfg-123".
+#
+# If using an external image registry, this step can be omitted, although you
+# will still need to create the push and pull secrets in the MCO namespace.
+oc get secrets -o name -n openshift-machine-config-operator -o=jsonpath='{.items[?(@.metadata.annotations.openshift\.io\/internal-registry-auth-token\.service-account=="builder")].metadata.name}'
+
+# (required): Create the layered MachineConfigPool:
+cat << EOF | oc create -f -
+---
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: layered
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - key: machineconfiguration.openshift.io/role
+        operator: In
+        values:
+        - worker
+        - layered
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/layered: ""
+EOF
+```
+
+### Customized OS Image
+
+We will use the following Containerfile to install some useful packages on our cluster nodes:
```

Comment:
Do we want to highlight the tree package here, as we use it later to verify? I missed this detail and didn't know why we were searching for the tree package later.

```suggestion
We will use the following Containerfile to install some useful packages on our cluster nodes, including the tree package, which we will use to verify the install later:
```